### ISSUE TITLE
allow retrieving game data from hash

### ISSUE THREAD

**Issue by Jamiras** [RetroAchievements/RAWeb#618]
As a client developer, I'd like to be able to fetch the achievements/leaderboards/rich presence data for a game using the same server call as resolving the hash.

Almost every time a hash is resolved via the `gameid` API, the next thing to do is to call the `patch` API with the resulting game ID.

I'd like to propose passing the hash to the `patch` API and have it return the patch data (which includes the game ID) if the hash resolves, or a null `PatchData` if the hash does not resolve (possibly even a Success=false,Error='Unknown Hash'). The API currently returns empty arrays when passed a non-existant game ID. This would also be acceptable as long as the `PatchData.ID` field is not set, it's obvious the game doesn't exist.
```
{"Success":true,"PatchData":{"Achievements":[],"Leaderboards":[]}}
```

This would avoid a second server request when loading a game, making it load just a little bit faster.

**Comment by luchaos** [comment]
Makes sense and should be easy to implement - in a backwards compatible fashion even. Ready for v1 anytime imo

**Comment by Jamiras** [comment]
Added in #2857
